scipy/stats/_continuous_distns.py

```diff
@@ -6540,10 +6540,90 @@ def _stats(self):
     def _entropy(self):
         return _EULER/2.0 + 1 - 0.5*np.log(2)
 
+    def fit(self, data, *args, **kwds):
+        """
+        Maxmimum likelihood estimation of the `scale` parameter
+        is used in the numerical optimization to determine `loc`.
+
+        Parameters
+        ----------
+        data : array_like
+            Data to use in calculating the maximum likelihood estimate.
+
+        Returns
+        -------
+        loc, scale : float
+            Maximum likelihood estimate for the scale, provided loc.
+
+        Notes:
+        If `floc` is fixed, the maximum likelihood scale may be calculated
+        without numerical optimization. Otherwise, numerical optimization
+        from `.rv_continuous` will be  conducted to determine `loc` and its
+        scale based on MLE.
+        """
+
+        floc = kwds.pop('floc', None)
+        fscale = kwds.pop('fscale', None)
+
+        def scale_mle(loc, data):
+            # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
+            # and Peacock (2000), Page 175
+            return (np.sum((data - loc) ** 2) / (2 * len(data))) ** .5
+
+        """
+        if `floc` is not fixed, provide a `_reduce_func` for `.rv_continuous`
+        to use. '_reduce_func' returns:
+            - the parameter to optimize from `x0`, loc
+            - `func` to minimize, and
+            - `get_tuple` to obtain the dependent parameter, as
+               numerical optimization over `func` only indicates
+               the independent parameter's solution.
+        """
+        check_fit_input_parameters(data, args, kwds,
+                                       fixed_param=(fscale, floc))
+        if floc is not None:
+            # if `floc` is fixed, determine `scale` with MLE
+            data = np.asarray(data)
+            return floc, scale_mle(floc, data)
+        if fscale is not None:
+            # `fscale` is fixed, use standard numerical optimization
+            return super(rayleigh_gen,
+                         self).fit(data, fscale=fscale)
+        else:
+            # both `floc` and `fscale` are free, use 
+            # custom `_reduce_func`
+            def _reduce_func(x0, ll, args, kwds):
+                def func(loc, data):
+                    return ll([loc, scale_mle(loc, data)], data)
+
+                def get_tuple(loc, data):
+                    return (loc.item(), scale_mle(loc, data))
+
+                return x0[0], func, get_tuple
+            return super(rayleigh_gen,
+                         self).fit(data, _reduce_func=_reduce_func)
 
 rayleigh = rayleigh_gen(a=0.0, name="rayleigh")
 
 
+def check_fit_input_parameters(data, args, kwds, fixed_param):
+    if len(args) > 0:
+        raise TypeError("Too many arguments.")
```

Comment:
Is this appropriate in general? Does this mean when using overridden `fit` method, only `data` and keyword arguments are acceptable?

Reply:
Corrected this to check
`len(args) > self.numargs`